Backpack/BarChart/Classes/BarChartCollectionView.swift

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+@objcMembers
+@objc(BPKBarChartCollectionView)
+public final class BPKBarChartCollectionView: UICollectionView {
+    
+    /// The BPKBarChartDataSource which will infom how the bar chart should be rendered
+    public var barChartDataSource: BPKBarChartCollectionViewDataSource?
+    
+    /// The BPKBarChartDelegate which can be used to respond to interaction with the bar chart
+    public weak var barChartDelegate: BPKBarChartCollectionViewDelegate?
+
+    /// The selected indexPath
+    var selectedIndexPath: IndexPath? {
+        didSet {
+            updateSelectedMarkerPosition()
+        }
+    }
+    
+    fileprivate static let cellIdentifier: String = "BPKBarChartCollectionView_CellIdentifier"
+    fileprivate var selectedMarkerBottomConstraint: NSLayoutConstraint?
+
+    public override init(frame: CGRect, collectionViewLayout layout: UICollectionViewLayout) {
+        super.init(frame: frame, collectionViewLayout: layout)
+        setupViews()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupViews()
+    }
+    
+    func setupViews() {
+        addSubview(selectedMarker)
+        
+        selectedMarkerBottomConstraint = selectedMarker.bottomAnchor.constraint(equalTo: bottomAnchor)
+        NSLayoutConstraint.activate([
+            selectedMarker.leadingAnchor.constraint(equalTo: frameLayoutGuide.leadingAnchor),
+            selectedMarker.trailingAnchor.constraint(equalTo: frameLayoutGuide.trailingAnchor),
+            selectedMarkerBottomConstraint!,
+            selectedMarker.heightAnchor.constraint(equalToConstant: 1.0)
+        ])
+        
+        register(BPKBarChartCollectionViewCell.self,
+                 forCellWithReuseIdentifier: BPKBarChartCollectionView.cellIdentifier)
+
+        collectionViewLayout = layout
+        dataSource = self
+        delegate = self
+        allowsSelection = true
+    }
+
+    lazy fileprivate var layout: UICollectionViewLayout = {
+        let layout = UICollectionViewFlowLayout()
+        layout.estimatedItemSize = CGSize(width: BPKSpacingXxl, height: bounds.height)
+        layout.scrollDirection = .horizontal
+        return layout
+    }()
+
+    lazy fileprivate var selectedMarker: UIView = {
+        let view = UIView()
+        view.backgroundColor = BPKColor.monteverde
+        view.isHidden = true
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    fileprivate func updateSelectedMarkerPosition() {
+        if selectedMarkerBottomConstraint != nil {
+            selectedMarkerBottomConstraint?.isActive = false
+            selectedMarkerBottomConstraint = nil
```

Comment:
Do we need to set this value to `nil` can't we just deactivate it and hide the view instead? This might be able save us from using a force unwrapped optional for the value?